src/librustdoc/clean/mod.rs

```diff
@@ -2247,16 +2255,17 @@ fn clean_maybe_renamed_item<'tcx>(
                 fields: variant_data.fields().iter().map(|x| clean_field(x, cx)).collect(),
             }),
             ItemKind::Impl(impl_) => return clean_impl(impl_, item.owner_id.def_id, cx),
-            // proc macros can have a name set by attributes
-            ItemKind::Fn(ref sig, generics, body_id) => {
-                clean_fn_or_proc_macro(item, sig, generics, body_id, &mut name, cx)
-            }
-            ItemKind::Macro(ref macro_def, _) => {
+            ItemKind::Macro(ref macro_def, MacroKind::Bang) => {
                 let ty_vis = cx.tcx.visibility(def_id);
                 MacroItem(Macro {
                     source: display_macro_source(cx, name, macro_def, def_id, ty_vis),
                 })
             }
+            ItemKind::Macro(_, macro_kind) => clean_proc_macro(item, &mut name, macro_kind, cx),
```

Comment:
So as far as I understand we treat non-bang macro declarations (`macro` or `macro_rules!`) as proc macros because we assume that they have `#[rustc_builtin_macro]` on them and since currently there is no way to define attributes and derive macros via declarative macros, right?

Seems fine to me, although that assumption may change in the far™ future.

Reply:
Nothing to link to, no. I faintly remember reading discussions about it someplace sometime.